src/NUnitFramework/framework/Api/NUnitTestAssemblyRunner.cs

```diff
@@ -244,6 +244,13 @@ public void StartRun(ITestListener listener)
             Context.Dispatcher = new SimpleWorkItemDispatcher();
 #endif
 
+            if (!System.Diagnostics.Debugger.IsAttached &&
+                Settings.Contains(PackageSettings.DebugTests) &&
+                (bool)Settings[PackageSettings.DebugTests])
```

Comment:
We should really write an `IDictionary` extension method for getting our settings to prevent the need for these double checks. Something like `static TReturn Value<TReturn>(this IDictionary, string key, TReturn default)`